### ISSUE TITLE
encountered some problems running train.py in PEM

### ISSUE THREAD

**Issue by jinguangle** [JiehongLin/SAM-6D#106]
sam6d) super@super-System-Product-Name:~/jin/SAM-6D-main/SAM-6D/Pose_Estimation_Model$ python train.py --gpus 0,1 --model pose_estimation_model --config config/base.yaml
/home/super/anaconda3/envs/sam6d/lib/python3.9/site-packages/gorilla/utils/gpu.py:6: FutureWarning: The pynvml package is deprecated. Please install nvidia-ml-py instead. If you did not install pynvml directly, please report this to the maintainers of the package that installed pynvml for you.
  import pynvml
set CUDA_VISIBLE_DEVICES as 0,1
2025-10-08 16:51:49,207 - ************************ Start Logging ************************
2025-10-08 16:51:49,207 - Config (path: config/base.yaml)
{
    "NAME_PROJECT": "SAM-6D",
    "optimizer": {
        "type": "Adam",
        "lr": 4e-05,
        "betas": [
            0.9,
            0.999
        ],
        "eps": 1e-06,
        "weight_decay": 0.05
    },
    "lr_scheduler": {
        "type": "WarmupCosineLR",
        "max_iters": 600000,
        "warmup_factor": 0.001,
        "warmup_iters": 1500
    },
    "model": {
        "coarse_npoint": 196,
        "fine_npoint": 2048,
        "feature_extraction": {
            "vit_type": "vit_base",
            "up_type": "linear",
            "embed_dim": 768,
            "out_dim": 256,
            "use_pyramid_feat": true,
            "pretrained": true
        },
        "geo_embedding": {
            "sigma_d": 0.2,
            "sigma_a": 15,
            "angle_k": 3,
            "reduction_a": "max",
            "hidden_dim": 256
        },
        "coarse_point_matching": {
            "nblock": 3,
            "input_dim": 256,
            "hidden_dim": 256,
            "out_dim": 256,
            "temp": 0.1,
            "sim_type": "cosine",
            "normalize_feat": true,
            "loss_dis_thres": 0.15,
            "nproposal1": 6000,
            "nproposal2": 300
        },
        "fine_point_matching": {
            "nblock": 3,
            "input_dim": 256,
            "hidden_dim": 256,
            "out_dim": 256,
            "pe_radius1": 0.1,
            "pe_radius2": 0.2,
            "focusing_factor": 3,
            "temp": 0.1,
            "sim_type": "cosine",
            "normalize_feat": true,
            "loss_dis_thres": 0.15
        }
    },
    "train_dataset": {
        "name": "training_dataset",
        "data_dir": "../Data/MegaPose-Training-Data",
        "img_size": 224,
        "n_sample_observed_point": 2048,
        "n_sample_model_point": 2048,
        "n_sample_template_point": 5000,
        "min_visib_fract": 0.1,
        "min_px_count_visib": 512,
        "shift_range": 0.01,
        "rgb_mask_flag": true,
        "dilate_mask": true
    },
    "train_dataloader": {
        "bs": 6,
        "num_workers": 8,
        "shuffle": true,
        "drop_last": true,
        "pin_memory": false
    },
    "test_dataset": {
        "name": "bop_test_dataset",
        "data_dir": "../Data/BOP",
        "template_dir": "../Data/BOP-Templates",
        "img_size": 224,
        "n_sample_observed_point": 2048,
        "n_sample_model_point": 1024,
        "n_sample_template_point": 5000,
        "minimum_n_point": 8,
        "rgb_mask_flag": true,
        "seg_filter_score": 0.25,
        "n_template_view": 42
    },
    "test_dataloader": {
        "bs": 16,
        "num_workers": 16,
        "shuffle": false,
        "drop_last": false,
        "pin_memory": false
    },
    "rd_seed": 1,
    "training_epoch": 15,
    "iters_to_print": 50,
    "exp_name": "pose_estimation_model_base_id0",
    "gpus": "0,1",
    "model_name": "pose_estimation_model",
    "log_dir": "log/pose_estimation_model_base_id0",
    "checkpoint_iter": -1
}
2025-10-08 16:51:49,208 - using gpu: 0,1
2025-10-08 16:51:49,208 - => creating model ...
load pre-trained checkpoint from: checkpoints/mae_pretrain_vit_base.pth
2025-10-08 16:51:51,699 - #Total parameters : 1027826136
Total 384000 images .....
2025-10-08 16:51:52,842 - 
Epoch 1 :
2025-10-08 16:51:57,762 - Iter 000001 Train - loss: 55.56387    coarse_loss0: 5.84276   coarse_loss1: 5.88709   coarse_loss2: 5.45751   coarse_acc: 0.16412     coarse_fg_num: 163.83334   coarse_dis: 1.11695     fine_loss0: 12.94693    fine_loss1: 12.81279    fine_loss2: 12.61678    fine_acc: 0.00000       fine_fg_num: 2048.00000 fine_dis: 0.98218  lr: 0.000000    T_data: 1.506   T_forward: 2.780        T_back: 0.631
2025-10-08 16:52:47,787 - Iter 000051 Train - loss: 52.18744    coarse_loss0: 5.52358   coarse_loss1: 5.44436   coarse_loss2: 5.24195   coarse_acc: 0.20820     coarse_fg_num: 155.18667   coarse_dis: 1.11193     fine_loss0: 12.25155    fine_loss1: 12.00121    fine_loss2: 11.72479    fine_acc: 0.00020       fine_fg_num: 2047.97000 fine_dis: 0.99762  lr: 0.000001    T_data: 0.236   T_forward: 0.431        T_back: 0.334
2025-10-08 16:53:38,181 - Iter 000101 Train - loss: 46.28374    coarse_loss0: 5.22061   coarse_loss1: 5.00382   coarse_loss2: 4.75776   coarse_acc: 0.31321     coarse_fg_num: 134.59334   coarse_dis: 1.10799     fine_loss0: 11.28923    fine_loss1: 10.44647    fine_loss2: 9.56585     fine_acc: 0.01208       fine_fg_num: 2023.87003 fine_dis: 1.00350  lr: 0.000002    T_data: 0.239   T_forward: 0.438        T_back: 0.331
2025-10-08 16:54:28,758 - Iter 000151 Train - loss: 36.63073    coarse_loss0: 4.86109   coarse_loss1: 4.49092   coarse_loss2: 4.32183   coarse_acc: 0.50509     coarse_fg_num: 96.93667    coarse_dis: 1.11039     fine_loss0: 9.57166     fine_loss1: 7.57829     fine_loss2: 5.80694     fine_acc: 0.72117       fine_fg_num: 570.66002  fine_dis: 0.70700  lr: 0.000003    T_data: 0.239   T_forward: 0.440        T_back: 0.333
2025-10-08 16:55:19,347 - Iter 000201 Train - loss: 25.54366    coarse_loss0: 4.38411   coarse_loss1: 4.04737   coarse_loss2: 4.00995   coarse_acc: 0.65201     coarse_fg_num: 67.79000    coarse_dis: 1.09568     fine_loss0: 7.65204     fine_loss1: 3.83278     fine_loss2: 1.61741     fine_acc: 0.98558       fine_fg_num: 28.94000   fine_dis: 0.04040  lr: 0.000005    T_data: 0.238   T_forward: 0.440        T_back: 0.333
2025-10-08 16:56:09,384 - Iter 000251 Train - loss: 17.87797    coarse_loss0: 4.05402   coarse_loss1: 3.72664   coarse_loss2: 3.70112   coarse_acc: 0.79543     coarse_fg_num: 39.75333    coarse_dis: 1.09587     fine_loss0: 5.34167     fine_loss1: 0.84769     fine_loss2: 0.20684     fine_acc: 0.99593       fine_fg_num: 6.82667    fine_dis: 0.00184  lr: 0.000006    T_data: 0.238   T_forward: 0.432        T_back: 0.330
2025-10-08 16:56:59,399 - Iter 000301 Train - loss: 13.25481    coarse_loss0: 3.69675   coarse_loss1: 3.33732   coarse_loss2: 3.20994   coarse_acc: 0.84964     coarse_fg_num: 29.18000    coarse_dis: 1.09572     fine_loss0: 2.66900     fine_loss1: 0.20055     fine_loss2: 0.14125     fine_acc: 0.99274       fine_fg_num: 13.65333   fine_dis: 0.00416  lr: 0.000007    T_data: 0.238   T_forward: 0.432        T_back: 0.330
2025-10-08 16:57:49,434 - Iter 000351 Train - loss: 10.22470    coarse_loss0: 3.40795   coarse_loss1: 2.94196   coarse_loss2: 2.72197   coarse_acc: 0.84823     coarse_fg_num: 29.67667    coarse_dis: 1.07246     fine_loss0: 0.97173     fine_loss1: 0.09496     fine_loss2: 0.08613     fine_acc: 0.99977       fine_fg_num: 0.00000    fine_dis: 0.00000  lr: 0.000009    T_data: 0.239   T_forward: 0.432        T_back: 0.330
2025-10-08 16:58:39,477 - Iter 000401 Train - loss: 8.30794     coarse_loss0: 3.05717   coarse_loss1: 2.42230   coarse_loss2: 2.07109   coarse_acc: 0.84789     coarse_fg_num: 29.42667    coarse_dis: 1.02768     fine_loss0: 0.43901     fine_loss1: 0.15909     fine_loss2: 0.15928     fine_acc: 0.98633       fine_fg_num: 27.30667   fine_dis: 0.00576  lr: 0.000010    T_data: 0.238   T_forward: 0.433        T_back: 0.329
2025-10-08 16:59:29,520 - Iter 000451 Train - loss: 5.67035     coarse_loss0: 2.46996   coarse_loss1: 1.55544   coarse_loss2: 1.17286   coarse_acc: 0.92080     coarse_fg_num: 15.43000    coarse_dis: 0.77017     fine_loss0: 0.25326     fine_loss1: 0.11063     fine_loss2: 0.10819     fine_acc: 0.99968       fine_fg_num: 0.00000    fine_dis: 0.00000  lr: 0.000011    T_data: 0.238   T_forward: 0.432        T_back: 0.330
2025-10-08 17:00:19,675 - Iter 000501 Train - loss: 2.66028     coarse_loss0: 1.37224   coarse_loss1: 0.62325   coarse_loss2: 0.40906   coarse_acc: 0.99641     coarse_fg_num: 0.54333     coarse_dis: 0.09011     fine_loss0: 0.15540     fine_loss1: 0.05084     fine_loss2: 0.04948     fine_acc: 0.99931       fine_fg_num: 0.00000    fine_dis: 0.00000  lr: 0.000013    T_data: 0.238   T_forward: 0.433        T_back: 0.331
2025-10-08 17:01:10,214 - Iter 000551 Train - loss: 1.62891     coarse_loss0: 0.62221   coarse_loss1: 0.29515   coarse_loss2: 0.22486   coarse_acc: 0.99942     coarse_fg_num: 0.00000     coarse_dis: 0.00000     fine_loss0: 0.19977     fine_loss1: 0.14418     fine_loss2: 0.14274     fine_acc: 0.99972       fine_fg_num: 0.00000    fine_dis: 0.00000  lr: 0.000014    T_data: 0.238   T_forward: 0.438        T_back: 0.334
2025-10-08 17:02:00,737 - Iter 000601 Train - loss: 0.65175     coarse_loss0: 0.26975   coarse_loss1: 0.13366   coarse_loss2: 0.09792   coarse_acc: 0.99963     coarse_fg_num: 0.00000     coarse_dis: 0.00000     fine_loss0: 0.09319     fine_loss1: 0.02885     fine_loss2: 0.02838     fine_acc: 0.99978       fine_fg_num: 0.00000    fine_dis: 0.00000  lr: 0.000015    T_data: 0.239   T_forward: 0.438        T_back: 0.333
2025-10-08 17:02:51,157 - Iter 000651 Train - loss: 1.07662     coarse_loss0: 0.22203   coarse_loss1: 0.16495   coarse_loss2: 0.14469   coarse_acc: 0.99440     coarse_fg_num: 0.00000     coarse_dis: 0.00000     fine_loss0: 0.20014     fine_loss1: 0.17374     fine_loss2: 0.17107     fine_acc: 0.99299       fine_fg_num: 13.65333   fine_dis: 0.00470  lr: 0.000017    T_data: 0.238   T_forward: 0.438        T_back: 0.332
2025-10-08 17:03:41,614 - Iter 000701 Train - loss: 0.67122     coarse_loss0: 0.13931   coarse_loss1: 0.10417   coarse_loss2: 0.08745   coarse_acc: 0.99944     coarse_fg_num: 0.00000     coarse_dis: 0.00000     fine_loss0: 0.13505     fine_loss1: 0.10343     fine_loss2: 0.10180     fine_acc: 0.99978       fine_fg_num: 0.00000    fine_dis: 0.00000  lr: 0.000018    T_data: 0.238   T_forward: 0.437        T_back: 0.333
2025-10-08 17:04:31,605 - Iter 000751 Train - loss: 0.72585     coarse_loss0: 0.14420   coarse_loss1: 0.11866   coarse_loss2: 0.10596   coarse_acc: 0.99631     coarse_fg_num: 0.00000     coarse_dis: 0.00000     fine_loss0: 0.13556     fine_loss1: 0.11105     fine_loss2: 0.11042     fine_acc: 0.99642       fine_fg_num: 6.82667    fine_dis: 0.00257  lr: 0.000019    T_data: 0.238   T_forward: 0.431        T_back: 0.329

**coarse_acc and fine_acc always high.    coarse_fg_num、coarse_dis、fine_fg_num and fine_dis drops quickly and remains at 0.**

**Is this normal?**
